whichone/static/js/main_module.js

```diff
@@ -0,0 +1,29 @@
+const view = {};
+const data = { user: null };
+const game = {};
+const controller = {};
+
+const initialisePage = function initialisePage(audioState, toggles) {
+  view.updateMuteIcon(audioState);
+  view.updateModeToggles(toggles);
+};
+
+// View module dependencies
+const inView = { document };
+View(inView, view);
+
+// Data module dependencies
+const inData = {
+  spotify_id: document.getElementById("info").getAttribute("data-user"),
+  initialisePage,
+};
+Data(inData, data);
+data.init();
+
+// Game module dependencies
+const inGame = { data: data, view: view };
+Game(inGame, game);
+
+// Controller module dependencies
+const inController = { game: game, data: data, view: view, document };
+Controller(inController, controller);
```

Comment:
I have a couple suggestions for the code design here.
1. There is no need to those functions to be uniform. I'd suggest to just pass game, data, view, document rather than wrapping it into a single object
2. Would be better if those functions were pure. Right now its a little strange to mutate the 2nd input as a form of "state". Maybe just return an object rather than mutating the input.
3. This might be useful - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze As I'd expect you don't want Controller to mutate View response for example.